hex/Cargo.toml

```diff
@@ -0,0 +1,33 @@
+[package]
+name = "fast-hex"
+version = "0.1.0"
+authors = ["Andrew Poelstra <apoelstra@wpsoftware.net>"]
+license = "CC0-1.0"
+description = "Fast hex encoding and decoding"
+homepage = "https://github.com/rust-bitcoin/hex/"
+repository = "https://github.com/rust-bitcoin/"
```

Comment:
`homepage` shouldn't point to the same URL as `repository` and `repository` shouldn't point to an organization.

Reply:
Damn, its hard to get good help huh :)

Reply:
I disagree with putting marketing words into crate descriptions, I mean it could be fast (and hope it is) but maybe not for every situation :).

Makes life easier when packaging for distributions that have guidelines on that sort of stuff, its hard to define what it's _fast_.

Reply:
Fair points, thanks man. I stayed away from `bitcoin-hex` because there is nothing in it that is specific to Bitcoin. The additional identifier is purely because `hex` is taken already. Will remove 'fast' though and fall back to `bitcoin-hex` unless we can come up with something better.

Reply:
Sorry didn't see the discussion on #1568 about other downstream crates needing a more up to date `hex` crate in which it would make sense to have a broader name. `bitcoin-hex` seems fine unless there's a better one.

Reply:
I also like `bitcoin-hex`. "What it has to do with bitcoin" is that it's maintained by the rust-bitcoin community, who has certain values about MSRV stability and minimizing dependency trees that others don't hold as strongly.

Reply:
Convincing enough for me :) `bitcoin-hex` it is.

Reply:
@Kixunil, this is the thread of discussion that led me to change the crate name to `bitcoin-hex`.

Reply:
How about something like `conservative-hex`/`solid-hex`...? Will pull GPT tomorrow to suggest more names if nobody beats me to it.

Reply:
I don't understand what's wrong with `bitcoin-hex` as a name for a hex project whose distinguishing feature is that it is mainained by the rust-bitcoin people, especially when multiple people are using our hex code specifically for that reason.

Reply:
Projects don't tend to be named after who maintains them. Ideally when one should get a rough idea what a crate is about by just reading the name (just like variables, types...)

There's also a marketing issue: if one used it in a non-bitcoin project it'd cause anyone unfamiliar looking at compilation output to say "why is this pulling in some unrelated bitcoin garbage?".

Reply:
I also like with what @jeandudey said someplace about not putting marketing terms in the crate name.

Reply:
We could also just use some silly short name like `hexify` or `hexer`.

Reply:
We *should* put *truthful* marketing names in the name. Saying `fast` is not truthful because we intentionally skip `unsafe` optimizations (which are possible as I demonstrated). Saying `conservative` is truthful because we have conservative MSRV and review process.

Truthful marketing is a good thing because it helps people who are searching for some solution to their problem identify the solution faster.

Reply:
> Projects don't tend to be named after who maintains them

They are named after the ecosystem that they exist in.

> There's also a marketing issue: if one used it in a non-bitcoin project it'd cause anyone unfamiliar looking at compilation output to say "why is this pulling in some unrelated bitcoin garbage?".

Lol. If people want to use actual garbage because they don't like our name, they're welcome to.

Reply:
>There's also a marketing issue: if one used it in a non-bitcoin project it'd cause anyone unfamiliar looking at compilation output to say "why is this pulling in some unrelated bitcoin garbage?".

I do use `bitcoin_hashes` even though it isn't explicitly related to Bitcoin.

It'd be better to keep a organization-like prefix so people know that if a crate starts with `bitcoin-.*` it is maintained by `rust-bitcoin` org.

Reply:
> `bitcoin_hashes`

False analogy, that crate specifically does contain things specific to Bitcoin - namely reversed `sha256d` and `hash160`. (Though I would've liked to have non-specific crate for algos alone.) Hex formatting is a universal concept and it cannot be possibly specific to Bitcoin.

>  crate starts with `bitcoin-.*` it is maintained by `rust-bitcoin` org.

That's very wrong and very dangerous assumption. Anyone can upload a crate starting with `bitcoin-`.

Anyway, if you all want it I won't block it even though I still dislike it.

Reply:
To me the name it should be is `hex` but we can't because of crates.io rules and everyone knows these rules so I always assumed everyone knew that the added fluff in names was mainly just that, fluff to get a unique name. Hence, I don't really care, hence I think `rust-hex` is perfectly adequate.

Reply:
> While it isn't the best scheme nor the most ergonomic

That's why I'm proposing to base the name on the description of the main goals: to maintain conservative MSRV and careful development. It's the same as with variables: clearer names are better. Consider the situation of someone looking at the dependency list of a crate unrelated to bitcoin. After seeing our crate he thinks:

* `hex-conservative-msrv` - "Ah, this crate tries to maintaing conservative MSRV"
* `bitcoin-hex` - "What does it have to do with bitcoin?" <spends a bunch of time investigating> "Oh, nothing, the crate just wants conservative MSRV"

If the naming doesn't matter why don't we just generate a random string and use that? Or name it `pink-pony`?

Reply:
>If the naming doesn't matter why don't we just generate a random string and use that? Or name it pink-pony?

For consistency